feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/detail/BalanceDetailViewModel.kt

```diff
@@ -126,6 +136,30 @@ class BalanceDetailViewModel(
         .inBackground()
         .share()
 
+    val priceChartTitle = assetFlow.map {
+        val tokenName = it.token.configuration.symbol.value
+        resourceManager.getString(R.string.price_chart_title, tokenName)
+    }
+
+    val priceChartFormatters: Flow<PriceChartTextInjectors> = flowOf {
+        val currency = currencyInteractor.getSelectedCurrency()
+
+        PriceChartTextInjectors(
+            RealPricePriceTextInjector(currency),
+            RealPriceChangeTextInjector(resourceManager, currency),
+            RealDateChartTextInjector(resourceManager)
+        )
+    }
+
+    private val priceCharts: Flow<List<AssetPriceChart>?> = assetFlow.flatMapLatest {
+        val priceId = it.token.configuration.priceId ?: return@flatMapLatest flowOf { null }
+        chartsInteractor.getCharts(priceId)
+    }
```

Comment:
We should not re-request the charts on each asset update - asset can be updated by many reasons, e.g price sync, balance sync
More efficient approach is to do something like `assetFlow.map { it.token.configuration.priceId }.distinctUntillChanged()` and use this flow as upstream